replace kube-core rendering with openshift operators render

mfojtik opened the pull request:
This change moves us away from kube-core renderer and use the kube-apiserver, kube-controller-manager and kube-scheduler operator renderers to gather all necessary bootstrap manifests. 

@abhinavdahiya @smarterclayton this is what we want and we should make working.

mfojtik commented:
FYI: Depends on https://github.com/openshift/cluster-kube-scheduler-operator/pull/16 https://github.com/openshift/cluster-kube-controller-manager-operator/pull/33 https://github.com/openshift/cluster-kube-controller-manager-operator/pull/32 and https://github.com/openshift/cluster-kube-apiserver-operator/pull/52

mfojtik commented:
/hold

mfojtik commented on pkg/asset/ignition/bootstrap/bootstrap.go:
```diff
@@ -24,8 +24,11 @@ import (
-	defaultReleaseImage = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	rootDir = "/opt/tectonic"
+
+	// TODO: This should be decided by installer, not hard-coded
```
@smarterclayton how are we supposed to handle this? :-) Also what is the preferred registry for our images? The RHCOS seems to have RH registry hardcoded as default, so using the `:latest` tag does not work and we really don't want to pin this into stable tag, otherwise we won't be able to test the changes in our operator images....

I guess we want all images in quay.io at some point but we want a knob to switch that to registry.svc.ci.openshift.org for the CI?

abhinavdahiya commented:
so the `kube-core-operator` installs these things in addition to api
```
1. kube-proxy
2. kube-dns
3. pod-checkpointer
```

What is the plan for these?

derekwaynecarr commented:
/cc @sjenning we need to discuss who shall install pod-checkpointer

mfojtik commented:
> 1. kube-proxy
> 2. kube-dns

/cc @ironcladlou

mfojtik commented on pkg/asset/tls/kubeletcertkey.go:
```diff
@@ -29,7 +29,9 @@ func (a *KubeletCertKey) Generate(dependencies asset.Parents) error {
 	cfg := &CertCfg{
-		Subject:      pkix.Name{CommonName: "system:serviceaccount:kube-system:default", Organization: []string{"system:serviceaccounts:kube-system"}},
+		// system:masters is a hack to get the kubelet up without kube-core
+		// TODO(node): make kubelet bootstrapping secure with minimal permissions eventually switching to system:node:* CommonName
```
/cc @sjenning

crawford commented:
/hold

This includes changes from https://github.com/openshift/installer/pull/44, which we believe is the wrong direction.

mfojtik commented:
@crawford sorry this PR should be marked as WIP. It is a state we want to be after we iron out all cert/path/infra problems in our operators. This PR in the end should include all changes needed in installer to make the openshift/kube control plane (bootstrap+daemonsets+operators) work.

mfojtik commented:
@crawford I added more detailed description for the changes in the commit messages, squashed the commits and removed the controversial commit. This is still work-in-progress, but should be more aligned with installer team contributing guidelines.

mfojtik commented:
/hold cancel

abhinavdahiya commented:
@mfojtik in resp to this https://github.com/openshift/installer/pull/420#issuecomment-430340074
https://github.com/openshift/installer/pull/420/commits/6b22542cd0f37071205a6ee76f920508adf18761
https://github.com/openshift/installer/pull/420/commits/e913a67856a6251c39317d8af4a918b9f3048819

are a no go.

sttts commented:
@abhinavdahiya we need an agreement how we proceed with https://github.com/openshift/installer/commit/2b6eae9cfdac07ec2b15500e269efc17ddbcdb3d and https://github.com/openshift/installer/commit/6b22542cd0f37071205a6ee76f920508adf18761. We all agree this is a no-go. On the other hand, both topics **block further control plane operator development**. I guess we as master team are fine with 
- either merging these temporary as is (and soon'ish follow-ups in the other operators), 
- or the node team (@sjenning) and the network team (@openshift/sig-networking) provide PRs against this branch to solve the situation before the big merge.

/cc @derekwaynecarr

sjenning commented:
For the checkpointer, @rphillips is working on an operator to do this.  But it is not required for an operational cluster.  We can drop it for the moment and add it back when Ryan is done.

sttts commented:
> We can drop it for the moment and add it back when Ryan is done.

Bootkube needs the checkpointer (a pod named like that) to proceed with bootstrapping.

deads2k commented:
/retest

abhinavdahiya commented:
reagrding https://github.com/openshift/installer/commit/2b6eae9cfdac07ec2b15500e269efc17ddbcdb3d i'm fine with this. Its not a blocker.

For https://github.com/openshift/installer/commit/6b22542cd0f37071205a6ee76f920508adf18761 I think a better and acceptable solution is going to be moving these manifests to installer assets like one here https://github.com/openshift/installer/blob/master/pkg/asset/manifests/operators.go

abhinavdahiya commented:
Also can we drop https://github.com/openshift/installer/commit/e913a67856a6251c39317d8af4a918b9f3048819 as that is already under discussion in #442

mfojtik commented:
@abhinavdahiya ok fixed. I don't know if the word "deprecated" is accurate, or we should use "temporary_". I tried to do my best to isolate that extra files for cleaner removal.

mfojtik commented on pkg/asset/ignition/bootstrap/content/bootkube.go:
```diff
@@ -121,15 +111,21 @@ then
 fi
 
+# TODO: Remove this when checkpointer and kube-proxy are properly rendered by their operators.
+echo "Installing temporary bootstrap manifests..."
```
@openshift/sig-pod @openshift/sig-networking this one is on you :-)

mfojtik replied:
/cc @derekwaynecarr

sttts commented on pkg/asset/ignition/bootstrap/content/bootkube_temporary.go:
```diff
@@ -0,0 +1,435 @@
+metadata:
+  labels:
+    k8s-app: kube-proxy
+    tectonic-operators.coreos.com/managed-by: kube-core-operator
```
nit: not true anymore

mfojtik commented:
@squeed @sjenning after this PR merges, I will spawn high prio BZ's to remove the temporary manifests and replace them with something that came out from your operators :-)

mfojtik commented:
/retest

mfojtik commented:
@smarterclayton @abhinavdahiya if I run the AWS cluster manually via installer, the nodes are all Ready, but in the CI job they are not. Any clue what might be different there than normal manual AWS install ?

mfojtik commented:
@smarterclayton also it seems like all container logs here are 0 size: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_installer/420/pull-ci-openshift-installer-master-e2e-aws/749/artifacts/e2e-aws/pods/

mfojtik commented:
/retest

abhinavdahiya commented:
kube-proxy on master:
```console
[core@adahiya-0-master-0 ~]$ sudo crictl logs 3df51a0b75309
W1017 13:08:32.060094       1 server.go:185] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
time="2018-10-17T13:08:32Z" level=warning msg="Running modprobe ip_vs failed with message: `modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/3.10.0-862.el7.x86_64/modules.dep.bin'\nmodprobe: WARNING: Module ip_vs not found in directory /lib/modules/3.10.0-862.el7.x86_64`, error: exit status 1"
time="2018-10-17T13:08:32Z" level=error msg="Could not get ipvs family information from the kernel. It is possible that ipvs is not enabled in your kernel. Native loadbalancing will not work until this is fixed."
I1017 13:08:32.071254       1 server_others.go:138] Using iptables Proxier.
W1017 13:09:02.072234       1 server.go:586] Failed to retrieve node info: Get https://10.3.0.1:443/api/v1/nodes/adahiya-0-master-0: dial tcp 10.3.0.1:443: i/o timeout
W1017 13:09:02.073122       1 proxier.go:463] invalid nodeIP, initializing kube-proxy with 127.0.0.1 as nodeIP
I1017 13:09:02.073684       1 server_others.go:171] Tearing down inactive rules.
I1017 13:09:02.132707       1 server.go:426] Version: v1.9.3+coreos.0
I1017 13:09:02.140133       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I1017 13:09:02.140189       1 conntrack.go:52] Setting nf_conntrack_max to 131072
E1017 13:09:02.140724       1 conntrack.go:124] sysfs is not writable: {Device:sysfs Path:/sys Type:sysfs Opts:[ro seclabel nosuid nodev noexec relatime] Freq:0 Pass:0} (mount options are [ro seclabel nosuid nodev noexec relatime])
I1017 13:09:02.140772       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I1017 13:09:02.140810       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I1017 13:09:02.141349       1 config.go:102] Starting endpoints config controller
I1017 13:09:02.141397       1 controller_utils.go:1019] Waiting for caches to sync for endpoints config controller
I1017 13:09:02.142410       1 config.go:202] Starting service config controller
I1017 13:09:02.142422       1 controller_utils.go:1019] Waiting for caches to sync for service config controller
E1017 13:09:32.141455       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:09:32.142039       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:09:32.142428       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:10:03.143028       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:10:03.144083       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:10:08.842063       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:10:34.144065       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:10:34.144971       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:10:48.843374       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:11:05.144705       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:11:05.145704       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:11:28.844335       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:11:36.151619       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:11:36.152695       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:12:07.152694       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:12:07.154048       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:12:08.845157       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:12:38.153952       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:12:38.155694       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:12:48.846127       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:13:09.154926       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:13:09.156374       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:13:28.847143       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:13:40.155831       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:13:40.157051       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:14:08.848530       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:14:11.156705       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:14:11.157759       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:14:42.159111       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:14:42.159113       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:14:48.848981       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:15:13.160106       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:15:13.160718       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:15:28.850026       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:15:44.161188       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:15:44.162222       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:16:08.851082       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:16:15.162272       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:16:15.163985       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:16:46.163138       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Endpoints: Get https://10.3.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:16:46.164769       1 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:85: Failed to list *core.Service: Get https://10.3.0.1:443/api/v1/services?limit=500&resourceVersion=0: dial tcp 10.3.0.1:443: i/o timeout
E1017 13:16:48.852640       1 event.go:209] Unable to write event: 'Post https://10.3.0.1:443/api/v1/namespaces/default/events: dial tcp 10.3.0.1:443: i/o timeout' (may retry after sleeping)
E1017 13:16:48.852775       1 event.go:144] Unable to write event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"adahiya-0-master-0.155e67561bee8771", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"Node", Namespace:"", Name:"adahiya-0-master-0", UID:"adahiya-0-master-0", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"readOnlySysFS", Message:"DOCKER RESTART NEEDED (docker issue #24000): /sys is read-only: cannot modify conntrack limits, problems may arise later.", Source:v1.EventSource{Component:"kube-proxy", Host:"adahiya-0-master-0"}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xbee9eafb8863bb71, ext:30293684095, loc:(*time.Location)(0xa9f7d80)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xbee9eafb8863bb71, ext:30293684095, loc:(*time.Location)(0xa9f7d80)}}, Count:1, Type:"Warning", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}' (retry limit exceeded!)
```

Also the apiserver to kubelet seems to be broken :?
```console
[6:40:33] ➜  dev git:(pr-420) ✗ oc -n kube-system logs -f tectonic-network-operator-wlpkvError from server: Get https://adahiya-0-master-0:10250/containerLogs/kube-system/tectonic-network-operator-wlpkv/tectonic-network-operator?follow=true: x509: certificate signed by unknown authority
```

deads2k commented:
I can reproduce the kubeconfig failure locally too.

mfojtik commented:
/retest

deads2k commented:
```
info: Manifests will be extracted to /tmp/release-image-2018-10-17T152741Z038656562
warning: Could not load current user information: user: unknown userid 1004640000
Unable to connect to the server: net/http: TLS handshake timeout
```

doesn't look like us

/retest

wking commented:
[e2e resource exhaustion][1]:

```
6 error(s) occurred:

* module.vpc.aws_eip.nat_eip[2]: 1 error(s) occurred:

* aws_eip.nat_eip.2: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: a976f054-6da1-4cae-9671-57af43fe96fe
* module.vpc.aws_eip.nat_eip[5]: 1 error(s) occurred:

...
```

I'm trying to reap orphaned resources in the CI account now.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/420/pull-ci-openshift-installer-master-e2e-aws/756/build-log.txt

deads2k commented:
/retest

abhinavdahiya commented:
https://github.com/openshift/machine-config-operator/pull/135 merged.

abhinavdahiya commented:
/retest

abhinavdahiya commented:
/test e2e-aws

abhinavdahiya commented:
tested locally on libvirt,
```console
Every 2.0s: oc get pods,nodes --all-namespaces                                                                                                                                                                                                           adahiya: Wed Oct 17 11:54:13 2018

NAMESPACE                              NAME                                                                  READY     STATUS              RESTARTS   AGE
default                                pod/registry-86f877866d-mw5hb                                         1/1       Running             0          7m
kube-system                            pod/kube-apiserver-tjz2m                                              1/1       Running             0          14m
kube-system                            pod/kube-controller-manager-9mbtx                                     1/1       Running             0          14m
kube-system                            pod/kube-dns-787c975867-skf2k                                         3/3       Running             0          14m
kube-system                            pod/kube-flannel-lkfqr                                                2/2       Running             2          5m
kube-system                            pod/kube-flannel-n2hkr                                                2/2       Running             0          12m
kube-system                            pod/kube-flannel-pqjgz                                                2/2       Running             0          12m
kube-system                            pod/kube-proxy-4zxfs                                                  1/1       Running             0          14m
kube-system                            pod/kube-proxy-d47p5                                                  1/1       Running             0          5m
kube-system                            pod/kube-proxy-jtdnr                                                  1/1       Running             0          14m
kube-system                            pod/kube-scheduler-pwdhd                                              1/1       Running             0          14m
kube-system                            pod/metrics-server-5767bfc576-m5llq                                   2/2       Running             0          8m
kube-system                            pod/pod-checkpointer-5bdd8                                            1/1       Running             0          14m
kube-system                            pod/pod-checkpointer-5bdd8-adahiya-0-master-0                         1/1       Running             0          13m
kube-system                            pod/tectonic-network-operator-rp94t                                   1/1       Running             0          14m
openshift-apiserver                    pod/apiserver-2r77k                                                   1/1       Running             0          10m
openshift-cluster-api                  pod/clusterapi-apiserver-6b855f7bc5-7b8tl                             2/2       Running             0          10m
openshift-cluster-api                  pod/clusterapi-controllers-85f6bfd9d5-s9kvv                           2/2       Running             0          9m
openshift-cluster-api                  pod/machine-api-operator-5d85454676-6kmw6                             1/1       Running             0          11m
openshift-cluster-samples-operator     pod/cluster-samples-operator-786df5bdf6-49nxx                         1/1       Running             0          7m
openshift-cluster-version              pod/cluster-version-operator-gkwbr                                    1/1       Running             0          14m
openshift-console                      pod/console-operator-75dbd98589-kg89j                                 1/1       Running             0          8m
openshift-controller-manager           pod/controller-manager-gn2xr                                          1/1       Running             0          10m
openshift-core-operators               pod/openshift-cluster-kube-apiserver-operator-769c6c55f8-4q5jt        1/1       Running             0          11m
openshift-core-operators               pod/openshift-cluster-kube-controller-manager-operator-8d9dcbd2gh5q   1/1       Running             0          11m
openshift-core-operators               pod/openshift-cluster-kube-scheduler-operator-5b47cbf44d-g7zpz        1/1       Running             0          11m
openshift-core-operators               pod/openshift-cluster-openshift-apiserver-operator-749cd6b5f6-fz6kc   1/1       Running             0          11m
openshift-core-operators               pod/openshift-cluster-openshift-controller-manager-operator-65jfpcb   1/1       Running             0          11m
openshift-core-operators               pod/openshift-service-cert-signer-operator-b84c9997d-dqnzj            1/1       Running             0          11m
openshift-image-registry               pod/cluster-image-registry-operator-5b57cc599d-nz5f4                  1/1       Running             0          7m
openshift-ingress                      pod/default-http-backend-6985d557bb-l7qqg                             1/1       Running             0          8m
openshift-ingress                      pod/router-845f8d6676-f277p                                           1/1       Running             0          7m
openshift-ingress                      pod/tectonic-ingress-controller-operator-fcb9c6f4b-bxfth              1/1       Running             0          9m
openshift-kube-apiserver               pod/apiserver-774754b557-5c7n8                                        1/1       Running             2          10m
openshift-kube-controller-manager      pod/controller-manager-895b7b7c5-9bbpm                                1/1       Running             0          10m
openshift-kube-scheduler               pod/scheduler-fc69c9cd9-4rdsv                                         1/1       Running             0          11m
openshift-machine-config-operator      pod/machine-config-controller-7f947c5bdf-4vw7d                        1/1       Running             0          9m
openshift-machine-config-operator      pod/machine-config-daemon-gn4wr                                       1/1       Running             0          5m
openshift-machine-config-operator      pod/machine-config-daemon-gscrt                                       1/1       Running             0          8m
openshift-machine-config-operator      pod/machine-config-operator-c4f595664-nxthv                           1/1       Running             0          11m
openshift-machine-config-operator      pod/machine-config-server-clm24                                       1/1       Running             0          9m
openshift-monitoring                   pod/cluster-monitoring-operator-c55d9b64c-dr2ns                       1/1       Running             0          7m
openshift-monitoring                   pod/grafana-57f595895d-ksms2                                          2/2       Running             0          1m
openshift-monitoring                   pod/prometheus-k8s-0                                                  0/4       ContainerCreating   0          17s
openshift-monitoring                   pod/prometheus-operator-5bf8644c75-6z9dp                              1/1       Running             0          2m
openshift-operator-lifecycle-manager   pod/catalog-operator-db57879b9-ctm88                                  1/1       Running             0          11m
openshift-operator-lifecycle-manager   pod/olm-operator-9dcc5b79c-r8dff                                      1/1       Running             0          11m
openshift-operator-lifecycle-manager   pod/package-server-d7c65d85c-7gvbg                                    0/1       CrashLoopBackOff    4          11m
openshift-service-cert-signer          pod/apiservice-cabundle-injector-b7cfb4749-vbvs9                      1/1       Running             0          10m
openshift-service-cert-signer          pod/configmap-cabundle-injector-78cd48cfcd-2nnsp                      1/1       Running             0          10m
openshift-service-cert-signer          pod/service-serving-cert-signer-69c7f47bf5-4b2ns                      1/1       Running             0          10m
openshift-web-console                  pod/webconsole-86f4f55644-qnwj9                                       1/1       Running             0          7m
tectonic-system                        pod/kube-addon-operator-784b4b6c7-jnm9l                               1/1       Running             0          10m

NAMESPACE   NAME                            STATUS    ROLES       AGE       VERSION
            node/adahiya-0-bootstrap        Ready     bootstrap   14m       v1.11.0+d4cacc0
            node/adahiya-0-master-0         Ready     master      14m       v1.11.0+d4cacc0
            node/adahiya-0-worker-0-q4jcr   Ready     worker      5m        v1.11.0+d4cacc0
```

/lgtm

deads2k commented:
job hung.  rekicking

/test e2e-aws

deads2k commented:
/test all

mfojtik commented:
/retest

mfojtik commented:
@abhinavdahiya reported that issue here: https://github.com/openshift/release/issues/1955

abhinavdahiya commented:
the e2e is failing [here](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/420/pull-ci-openshift-installer-master-e2e-aws/776)

```console
Error: Error applying plan:

6 error(s) occurred:

* module.vpc.aws_eip.nat_eip[5]: 1 error(s) occurred:

* aws_eip.nat_eip.5: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: c512d767-2493-4812-9561-5d1197723cf7
* module.vpc.aws_eip.nat_eip[0]: 1 error(s) occurred:

* aws_eip.nat_eip.0: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: cac50c99-211c-49fc-9204-1c9e99f03ae7
* module.vpc.aws_eip.nat_eip[3]: 1 error(s) occurred:

* aws_eip.nat_eip.3: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: 77877ad8-d126-4850-894c-01e31861aa08
* module.vpc.aws_eip.nat_eip[4]: 1 error(s) occurred:

* aws_eip.nat_eip.4: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: 4ab4d76f-ae87-41d5-b51e-4905882fe710
* module.vpc.aws_eip.nat_eip[1]: 1 error(s) occurred:

* aws_eip.nat_eip.1: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: 952b9a6f-5742-4d1d-87ea-44f5b80d796e
* module.vpc.aws_eip.nat_eip[2]: 1 error(s) occurred:

* aws_eip.nat_eip.2: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
	status code: 400, request id: aaaed3f9-64d7-4d82-975c-81eeda85da7e

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

I am going to ask @wking to cleanup the CI account. :(

abhinavdahiya commented:
/test e2e-aws

abhinavdahiya commented:
/retest

wking commented:
The [previous e2e-aws error][1]

```
2018/10/17 22:42:09 Executing template e2e-aws
2018/10/17 22:42:09 Creating or restarting template instance
2018/10/17 22:42:09 Waiting for template instance to be ready
2018/10/17 22:42:12 Ran for 2m5s
error: could not run steps: could not wait for template instance to be ready: could not determine if template instance was ready: failed to create objects: object is being deleted: pods "e2e-aws" already exists
```

That's openshift/release#1955.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/420/pull-ci-openshift-installer-master-e2e-aws/780/build-log.txt

deads2k commented:
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/420/pull-ci-openshift-installer-master-e2e-aws/781

```
/tmp/openshift/build-rpms/rpm/BUILD/origin-4.0.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:684
Expected error:
    <*errors.errorString | 0xc420a0f160>: {
        s: "failed to get logs from pod-secrets-69066412-d263-11e8-bde6-0a58ac1018e2 for secret-volume-test: an error on the server (\"unknown\") has prevented the request from succeeding (get pods pod-secrets-69066412-d263-11e8-bde6-0a58ac1018e2)",
    }
    failed to get logs from pod-secrets-69066412-d263-11e8-bde6-0a58ac1018e2 for secret-volume-test: an error on the server ("unknown") has prevented the request from succeeding (get pods pod-secrets-69066412-d263-11e8-bde6-0a58ac1018e2)
not to have occurred
/tmp/openshift/build-rpms/rpm/BUILD/origin-4.0.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:2351
```

@sttts @mfojtik maybe we hadn't gotten this far before?

deads2k commented:
https://github.com/openshift/origin/pull/21286/files merged.  Missing the operator half maybe?

sttts commented:
/lgtm

sttts commented on pkg/asset/ignition/bootstrap/content/bootkube.go:
```diff
@@ -232,3 +231,30 @@ podman run \
+apiVersion: kubecontrolplane.config.openshift.io/v1
+kind: KubeAPIServerConfig
+kubeletClientInfo:
+  ca: "" # kubelet uses self-signed serving certs. TODO: fix kubelet pki
```
@sjenning this is for you. We have self-signed kubelet serving certs right now. I.e. the kube-apiserver->kubelet communication is insecure because the kubelet identity is not checked.

sttts replied:
This is the old kube-core behaviour btw. So we are just copying that here, but should do better. I read @liggit commenting somewhere that up to kube 1.10 we don't support CSRs for properly signed kubelet serving certs. Has this changed upstream in 1.11 or 1.12?

mfojtik replied:
https://github.com/openshift/installer/issues/486

sttts commented on pkg/asset/ignition/bootstrap/content/bootkube.go:
```diff
@@ -229,3 +231,30 @@ podman run \
+	KubeControllerManagerConfigOverridesTemplate = template.Must(template.New("kube-controller-manager-config-overrides.yaml").Parse(`
+apiVersion: kubecontrolplane.config.openshift.io/v1
+kind: KubeControllerManagerConfig
+`))
```
@squeed this file is also for you. You can override any setting related to networking here, like cluster CIDR for example.
